### PR TITLE
Feature/export csv nuget package

### PR DESCRIPTION
**Pull request by PinkhamSoftware**
What this does:
- Return a List of AssetOutputModels when the accept: text/csv header is present.
- Serialize to CSV format using a nuget package which serializes all fields in order that they appear in the class.
- Prompts to download in the browser.
- Currently filename is simply export.csv
- Correctly opens in Excel by default.
screen shot 2018-12-05 at 08 53 58
- Allows to download CSV serialized list as text.
screen shot 2018-12-05 at 08 50 45

What this needs to do:
- Add End to End tests to test serialization.

Potential issues:
- Control over formatting of dates.

Potential improvements:
- Set filename to be more contextual.

### CODE REVIEW

**Comment by danielburnley** [comment]
(discussed IRL - noting here also)

It'd be good to be able to get this wrapped in some tests that actually check the CSV generated - Although we're using a library we still need to get this tested so that we can have confidence in our dependency updates.

It'd be good to avoid testing through the entire stack, as we only want to test up the the boundaries of our delivery mechanism - one way of doing this is similar to how we currently test controllers, which is by mocking out the usecases. However, as we need to send a real request and check the response we'll need to spin up the entire application. 

If possible it'd be good if we could swap out the dependency injection so that it uses stubbed values to maintain separation of the two boundaries - we can then spin up the API and send it a request, and then parse the CSV that was downloaded.

**Comment by PinkhamSoftware** [comment]
Yep agreed, my current train of thought is to create an EndToEndTest solution where we spin up the whole API in memory, mock out the dependencies and serve http requests to it. This however will make this PR quite large.

File: ArchitectureDecisionRecords/1-CsvExport.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+**Status**: Accepted
+
+**Context**: The client use case is exporting indivual records as comma separated file (.csv)
+- Option 1: Manually serialize the data to csv and output it via the ```Response.WriteAsync();```
+- Option 2: Create Custom OutputFormatter 
+- Option 3: Use existing nuget package.
+
+**Decision**: The decision we have gone with is Option 3. Thereby removing the need to have the output model be aware of any sort of serialization, and allowing us quickest and cleanest delivery time of this feature.
+
+**Consequences**:
```
Comment (danielburnley) [inline ArchitectureDecisionRecords/1-CsvExport.md]: I'd be interested in a standard of how we should write the consequences - but for this PR and an initial attempt at the ADRs this works

Comment (PinkhamSoftware) [reply ArchitectureDecisionRecords/1-CsvExport.md]: Yes, this definitely needs iteration. Once we've agreed and worked out a process/workflow then for the moment this is a first cut of the final product.

**Pull request merged**